src/System.Windows.Forms/src/System/Windows/Forms/DragDropHelper.cs

```
@@ -0,0 +1,562 @@
+                dragSourceHelper = (IDragSourceHelper2)obj;
+                return true;
+            }
+            catch
+            {
+                dragSourceHelper = null;
+                return false;
+            }
+        }
+
+        /// <summary>
+        ///  Creates an in-process server drag-image manager object and returns an interface pointer to
+        ///  IDropTargetHelper. Exposes methods that allow drop targets to display a drag image while the image is over
+        ///  the target window.
+        /// </summary>
+        private static bool TryGetDropTargetHelper([NotNullWhen(true)] out IDropTargetHelper? dropTargetHelper)
+        {
+            try
+            {
+                Ole32.CoCreateInstance(
+                    ref CLSID.DragDropHelper,
+                    IntPtr.Zero,
+                    Ole32.CLSCTX.INPROC_SERVER,
+                    ref NativeMethods.ActiveX.IID_IUnknown,
+                    out object obj).ThrowIfFailed();
+                dropTargetHelper = (IDropTargetHelper)obj;
+                return true;
+            }
+            catch
+            {
+                dropTargetHelper = null;
+                return false;
+            }
+        }
+
+        /// <summary>
+        ///  Notifies the drag-image manager that the drop target has been entered, and provides it with the information
+        ///  needed to display the drag image.
+        /// </summary>
+        public static void DragEnter(IntPtr hwndTarget, IComDataObject dataObject, ref Point ppt, uint dwEffect)
+        {
+            if (!TryGetDropTargetHelper(out IDropTargetHelper? dropTargetHelper))
+            {
+                return;
+            }
+
+            try
+            {
+                dropTargetHelper.DragEnter(hwndTarget, dataObject, ref ppt, dwEffect).ThrowIfFailed();
+            }
+            catch
+            {
+                return;
+            }
+            finally
```

nit: empty catch block